vecvec-lapack/tests/TST/Memory.hs

```diff
@@ -0,0 +1,59 @@
+module TST.Memory (tests) where
+
+
+import           Control.Monad.ST
+import           Data.Foldable                         (for_)
+import qualified Data.Vector.Generic                   as VG
+import qualified Data.Vector.Generic.Mutable           as MVG
+
+import           Vecvec.Classes.NDArray                (Length (..), slice)
+import qualified Vecvec.LAPACK                         as VV
+import           Vecvec.LAPACK.Internal.Vector.Mutable (Strided (..))
+
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+
+tests :: TestTree
+tests = testGroup "memory"
```

Comment:
Test could be simplified. Specification for move is quite simple: If vectors overlap, the copying is performed as if the source vector were copied to a temporary vector and then the temporary vector was copied to the target vector. So two program should be equivalent:

```haskell
carrier <- MVG.generateM carrierLength id
let source =  slice ... carrier
let target =  slice ... carrier
MVG.move target source
VG.unsafeFreeze carrier
```

```haskell
carrier <- MVG.generateM carrierLength id
let source =  slice ... carrier
let target =  slice ... carrier
buf <- MVG.clone source
MVG.copy target buf
VG.unsafeFreeze carrier
```

Reply:
Done